### ISSUE TITLE
Misleading example in Deferred Intro

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7881]
|<img alt="elpollodiablo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| elpollodiablo reported|
|-|-|
|Trac ID|trac#7881|
|Type|defect|
|Created|2015-05-02 15:56:25Z|
|Branch|https://github.com/twisted/twisted/tree/doc-howto-defer-7881|

https://twistedmatrix.com/documents/15.1.0/core/howto/defer.html#callbacks

uses d misleadingly as an argument to printData() which does in fact in this example receive an actual value, not an instance of Deferred for which d is used throughout the examples. getDummyData() is also getting a discrete value, hence x is used, not d.

This has been an unnecessary source of confusion for me. Continuity in naming is not unimportant in code examples.

Attachments:

* [defer-example.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3bd/3bd5717abb673d0c26b7f6356e86610bad746385/d55f6766c4633684886e61eb6c90a0d360458d82.patch) (3899 bytes) - added by thijs on 2015-11-18 17:13:16Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7881 7881
type__defect defect
reporter__elpollodiablo elpollodiablo
priority__low low
milestone__ 
branch__branches_doc_howto_defer_7881 branches/doc-howto-defer-7881
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1430582185019222 1430582185019222
changetime__1447958636626015 1447958636626015
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

Attached patch updates the var names and fixes old-style instances of `print` throughout the document.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

(In [46266]) Branching to doc-howto-defer-7881.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @thijstriemstra|
|-|-|

Many thanks for the changes.

Since I was reading this I have also done a few other renames... basically converting someFunction to cbSomeFunction or ebSomeFunction to signal that the code should not be called directly and it is expected to be called as a callback/errback

Will run the test and merge.

Please check that my changes make sense.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra set owner to @adiroiban|
|-|-|

Thanks. There's one new typo: `print('getDummyData caleed')`. When you fixed that this should be good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @thijstriemstra|
|-|-|

Thanks! Fixed.. Will merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set status to `closed`|
|-|-|

(In [46293]) Merge doc-howto-defer-7881: Fix deferred docs examples.

Author: thijs
Reviewer: adiroiban
Fixes: [#7881](https://github.com/twisted/twisted/issues/7881)
